packages/frontend/src/components/UserSettings/AccessTokensPanel/index.tsx

```diff
@@ -109,48 +19,38 @@ const AccessTokensPanel: FC = () => {
         );
     }
 
-    return (
-        <Flex
-            h="100%"
-            direction="column"
-            justify={hasAvailableTokens ? 'auto' : 'center'}
-            align={hasAvailableTokens ? 'auto' : 'center'}
-        >
-            {hasAvailableTokens && (
-                <HeaderActions>
-                    <PanelTitle>Personal access tokens</PanelTitle>
-                    <Button
-                        intent="primary"
-                        onClick={() => setCreateInvitesPanel(true)}
-                        text="Generate new token"
+    if (!hasAvailableTokens) {
+        return (
+            <EmptyState
+                icon={
+                    <MantineIcon
+                        icon={IconKey}
+                        color="gray.6"
+                        stroke={1}
+                        size="5xl"
                     />
-                </HeaderActions>
-            )}
-            {hasAvailableTokens ? (
-                <div>
-                    {data?.map((token) => (
-                        <TokenListItem key={token.uuid} token={token} />
-                    ))}
-                </div>
-            ) : (
-                <EmptyState
-                    icon={
-                        <MantineIcon
-                            icon={IconKey}
-                            color="gray.6"
-                            stroke={1}
-                            size="5xl"
-                        />
-                    }
-                    title="No tokens"
-                    description="You haven't generated any tokens yet!, generate your first token"
-                >
-                    <MantineButton onClick={() => setCreateInvitesPanel(true)}>
-                        Generate token
-                    </MantineButton>
-                </EmptyState>
-            )}
-        </Flex>
+                }
+                title="No tokens"
+                description="You haven't generated any tokens yet!, generate your first token"
+            >
+                <Button onClick={() => setCreateInvitesPanel(true)}>
+                    Generate token
+                </Button>
+            </EmptyState>
+        );
+    }
+
+    return (
+        <>
+            <Flex justify="space-between" align="center" mb="lg">
```

Comment:
you can use `Group position="apart"` and you don't need align anymore.

also, not a big fan of applying margins to one element to have a gap between two separate sections.

I would wrap the whole content inside a <Stack spacing='lg'>

Reply:
that's great, thank you! Wrapped these 2 with a Stack, and applied mb to the Stack!